src/polyfills/set.js

```diff
@@ -14,19 +14,35 @@
  * limitations under the License.
  */
 
+import {toArray} from '#core/types/array';
+
 /**
- * Forces the return value from Set.prototype.add to always be the set
- * instance. IE11 returns undefined.
- *
  * @param {!Window} win
  */
 export function install(win) {
-  const {Set} = win;
-  const s = new Set();
+  const {Set: SetConstructor} = win;
+  const s = new SetConstructor([1]);
+  // Add suppport for `new Set(iterable)`. IE11 lacks it.
+  if (s.size < 1) {
+    win.Set = /** @type {typeof Set} */ (
+      function (iterable) {
+        const set = new SetConstructor();
+        if (iterable) {
+          const asArray = toArray(iterable);
+          for (let i = 0; i < asArray.length; i++) {
+            set.add(asArray[i]);
```

Comment:
In order to support iterables, and not just array-likes.

```suggestion
          for (const it of iterable) {
            set.add(it);
```